Старовая реализация сервиса x6user

KiryakovDenis opened the pull request.

KiryakovDenis committed:
Старовая реализация сервиса x6user

gmnpr commented on src/main/java/ru/kdv/study/config/OpenApiConfig.java:
```diff
@@ -0,0 +1,20 @@
+        )
+)
+public class OpenApiConfig {
+    // Конфигурация для Swagger
```
классы и методы комментируются над объявлением. вообще насколько такой комментарий полезный?)

gmnpr commented on src/main/java/ru/kdv/study/controller/UserExistController.java:
```diff
@@ -0,0 +1,23 @@
+@RestController
+@RequestMapping("/api/v1/exist")
+@RequiredArgsConstructor
+public class UserExistController {
```
почему это другой контроллер?

gmnpr commented on src/main/java/ru/kdv/study/model/ServiceError.java:
```diff
@@ -0,0 +1,10 @@
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
```
@Data?

gmnpr commented on src/main/java/ru/kdv/study/repository/UserRepository.java:
```diff
@@ -0,0 +1,112 @@
+        }
+    }
+
+    public Boolean userExist(Long id) {
```
Boolean? может быть состояние null?

gmnpr commented on src/main/java/ru/kdv/study/repository/UserRepository.java:
```diff
@@ -0,0 +1,112 @@
+
+    public Boolean userExist(Long id) {
+        try {
+            return jdbcTemplate.queryForObject(SELECT_BY_EXIST, new MapSqlParameterSource("id", id), userExistMapper);
```
jdbcTemplate.queryForObject(SELECT_BY_EXIST, new MapSqlParameterSource("id", id), userExistMapper, Boolean.class)

gmnpr commented on src/main/java/ru/kdv/study/repository/UserRepository.java:
```diff
@@ -0,0 +1,112 @@
+
+    private static String UPDATE = """
+            UPDATE user_service."user"
+               SET name = :name
```
а почту?)

gmnpr commented on src/main/java/ru/kdv/study/service/UserService.java:
```diff
@@ -0,0 +1,58 @@
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private static String REGEX_EMAIL_PATTERN_RFC5322 = "^[a-zA-Z0-9_!#$%&'*+/=?`{|}~^.-]+@[a-zA-Z0-9.-]+$";
```
у констант еще есть final

gmnpr commented on src/main/java/ru/kdv/study/service/UserService.java:
```diff
@@ -0,0 +1,58 @@
+        if (!StringUtils.hasText(user.getName())) {
+            throw BadRequestException.create("Поле name должно быть заполнено.");
+
+        } else if (!StringUtils.hasText(user.getEmail())) {
```
зачем else?) сделай отдельные ифы, будет сильно читаемей

gmnpr commented on src/main/resources/db/changelog/202502211628__init_tables.sql:
```diff
@@ -0,0 +1,8 @@
+    id SERIAL PRIMARY KEY,
+    name VARCHAR NOT NULL,
+    email VARCHAR NOT NULL,
+    create_date TIMESTAMP WITHOUT TIME ZONE NOT NULL
```
давай изменим create_date, чтобы поле само себя заполняло во время создания новой записи

KiryakovDenis committed:
Доработки по ревью

gmnpr commented on src/main/java/ru/kdv/study/service/UserService.java:
```diff
@@ -0,0 +1,60 @@
+            throw BadRequestException.create("Поле email должно быть заполнено.");
+        }
+
+        if (!emailIsValid(user.getEmail(), REGEX_EMAIL_PATTERN_RFC5322)) {
```
а зачем регулярка передается как аргумент?) могут быть разные  реглуярки для валидации почт? пусть сразу будет в методе emailIsValid

KiryakovDenis committed:
Доработки по ревью.
Шаблон регулярного выражения перенесен внутрь метода emailIsValid в качестве локальной переменной

gmnpr commented on src/main/java/ru/kdv/study/model/ServiceError.java:
```diff
@@ -0,0 +1,10 @@
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@AllArgsConstructor
```
точно нужно?

gmnpr replied:
мне кажется везде хватит просто @Data 
проверь все модели

gmnpr commented on src/main/java/ru/kdv/study/controller/UserController.java:
```diff
@@ -0,0 +1,45 @@
+
+    @PostMapping("user/exist/{id}")
+    @Operation(summary = "Проверить наличие пользователя по id")
+    public Boolean userExists(@PathVariable final Long id) {
```
точно Boolean?

gmnpr commented on src/main/java/ru/kdv/study/repository/UserRepository.java:
```diff
@@ -0,0 +1,108 @@
+    public boolean userExist(Long id) {
+        try {
+            return Boolean.TRUE.equals(jdbcTemplate.queryForObject(SELECT_BY_EXIST, new MapSqlParameterSource("id", id), Boolean.class));
+
```
зачем пустая строка?

gmnpr commented on src/main/java/ru/kdv/study/service/UserService.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    @Transactional(readOnly = true)
+    public Boolean userExist(final Long id) {
```
точно Boolean?

gmnpr commented on src/main/java/ru/kdv/study/service/UserService.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    private boolean emailIsValid(String email) {
+        String emailPatterRFC3522 = "^[a-zA-Z0-9_!#$%&'*+/=?`{|}~^.-]+@[a-zA-Z0-9.-]+$";
```
константа

gmnpr commented on src/main/java/ru/kdv/study/config/OpenApiConfig.java:
```diff
@@ -0,0 +1,19 @@
+                version = "1.0.0",
+                contact = @Contact(
+                        name = "Kiryakov Denis",
+                        email = "KiryakovDenis@gmail.com"
```
щас боты проиндексируют будут тебе приветы передавать

gmnpr commented on src/main/java/ru/kdv/study/repository/UserRepository.java:
```diff
@@ -0,0 +1,108 @@
+@Repository
+@RequiredArgsConstructor
+public class UserRepository {
+    private static String INSERT = """
```
константа это static + final (видимо надо еще отдельно писать, что поправить надо во всем проекте :D)

gmnpr commented on src/main/java/ru/kdv/study/repository/UserRepository.java:
```diff
@@ -0,0 +1,108 @@
+public class UserRepository {
+    private static String INSERT = """
+            INSERT 
+              INTO user_service."user" ("name", email) 
```
зачем перенос строки? 
INSERT INTO user_service."user" ("name", email)
...

KiryakovDenis replied:
Это пальцы сами так пишут. Стайл гайд по оформлению запросов в базе.

gmnpr commented on src/main/java/ru/kdv/study/repository/UserRepository.java:
```diff
@@ -0,0 +1,108 @@
+import ru.kdv.study.model.User;
+import ru.kdv.study.repository.mapper.UserMapper;
+
+import java.time.LocalDateTime;
```
а это зачем?)

gmnpr commented on src/main/java/ru/kdv/study/exception/ExceptionController.java:
```diff
@@ -0,0 +1,46 @@
+
+    @ExceptionHandler(NoDataFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public ResponseEntity<ServiceError> hadleNoDataFoundException(NoDataFoundException e) {
```
hadle -> handle

KiryakovDenis committed:
Доработки по ревью
Boleean -> boolean
Пустую строку удалил
В моделях везде @Data + @AllArgConstructor

KiryakovDenis committed:
Почту поменял

KiryakovDenis committed:
Удалил неиспользуемый импорт.
Исправил hadle->handle

KiryakovDenis committed:
Удалил неиспользуемый импорт.
Исправил hadle->handle
Исправил форматирование INSERT
static -> static final исправил везде, вроде бы.

KiryakovDenis committed:
Добавлено кэширование

KiryakovDenis committed:
Удалил папку gradle

KiryakovDenis committed:
Допилил gitignore

KiryakovDenis committed:
Интерфейс serializable для модели User

KiryakovDenis committed:
Перенес настройку Swagger из конфигурационного файла в properties.yaml,
Подправил лямбды,
Подправил codestyle

KiryakovDenis committed:
Удалены файлы gradlew и gradlew.bat

KiryakovDenis committed:
Изменен .gitignore

gmnpr commented on src/main/java/ru/kdv/study/repository/UserRepository.java:
```diff
@@ -0,0 +1,104 @@
+    private RuntimeException handleDbExceptionMessage (final Exception e, final User user) throws DataBaseException{
+        if (e instanceof EmptyResultDataAccessException) {
+            return NoDataFoundException.create(String.format("Пользователь не найден {id = %s}", user.getId()));
+        } if (e.getMessage().contains("user_uk1")) {
```
code style
если это обычный if, то с новой строки

gmnpr commented on src/main/java/ru/kdv/study/model/User.java:
```diff
@@ -0,0 +1,16 @@
+
+@Data
+@AllArgsConstructor
+public class User implements Serializable {
```
давай без Serializable
